gridcomps/Cap/MAPL_Cap.F90

```diff
@@ -419,10 +420,9 @@ subroutine finalize_mpi(this, unusable, rc)
       integer :: status
       _UNUSED_DUMMY(unusable)
 
+      call MAPL_Finalize(comm=this%comm_world)
```

Comment:
This call is illegal if MPI is already finalized - the comm may no longer exist.  So either there is a problem here or the conditional on the next line is unnecessary.

Reply:
I don't think MPI_finalize should have been called at this point.  If MPI_Initialize is called by Cap's parents, then it is its parents' responsibility to call MPI_finalize() after our Cap's  finalize_mpi() call.

Reply:
I'm now realizing that I had misinterpreted the conditional that follows this line.